modules/index/src/main/java/org/apache/ignite/internal/index/IndexManager.java

```diff
@@ -64,7 +61,7 @@
  * as well as managing indexes' lifecycle.
  */
 // TODO: IGNITE-19082 Delete this class
-public class IndexManager extends AbstractEventProducer<IndexEvent, IndexEventParameters> implements IgniteComponent {
+public class IndexManager implements IgniteComponent {
```

Comment:
There is no need to get rid of **AbstractEventProducer** it will be needed for IGNITE-20330

Reply:
Is there any arh document to see how it will be used?

Reply:
@ygerzhedovich I didn’t quite understand why an architectural document was needed, I always thought it didn’t go deep into the implementation but revealed a high-level algorithm.

Reply:
As I see IGNITE-20330 is opened, but has a PR with concern from @ibessonov about architecture [1]. Also, I see IndexEvent is not used in PR.

Usually, Events is used to notify external components when state was changed.
Here, IndexManager is fully responsible for index building flow. What external component needs subscribing to the IndexEvent?
As I see, Catalog can't do that because of position in dependency graph, it just know nothing about IndexManager. IndexBuilder looks like internal component relating to IndexManager, and I have no idea why it passed from the outside (however, this can be a IndexBuilder factory though). Thus, IndexManagere can call builder methods directly.

[1] https://github.com/apache/ignite-3/pull/2538/files#r1317100029

Reply:
> As I see IGNITE-20330 is opened, but has a PR with concern from @ibessonov about architecture [1]. Also, I see IndexEvent is not used in PR.

I closed the [PR](https://github.com/apache/ignite-3/pull/2538) because it is no longer valid because there will be a new implementation. We personally discussed (with @ibessonov and @sanpwc) the past decision and came to a new decision that will use the `IndexEvent` and `PrimaryReplicaEvent`.

> Usually, Events are used to notify external components (e.g. which we can't foresee in constructor) when state was changed.
> 
> Here, IndexManager is fully responsible for index building flow. What external component needs subscribing to the IndexEvent?

A new component `IndexBuildController` will appear, which will run the index building procedure, it will listen to events from `IndexManager` and `PlacementDriver` and react correctly to the its events.

This is necessary firstly for less influence on the code of other components and also for merging two types of events:
1. Index appeared/deleted.
2. Appearance/loss of primacy of a replica.

Reply:
Moreover, I think IgniteManager is a IgniteComponent in the first place, and not an event Producer. The current approach with extending Producer class is wrong.

If we decide at some point got extract common component behaviour into an abstract class, then a refactoring must be done. Let's rewrite event processing using incapsulation instead of inherintance.

Reply:
> Thus, IndexManager can just call builder methods directly.

This is precisely my concern, I don't want index manager to depend on our internal stuff. Events allows us to remove such unnecessary dependencies.

Reply:
> Moreover, I think IgniteManager is a IgniteComponent in the first place, and not an event Producer. The current approach with extending Producer class is wrong.

Why wrong?

> If we decide at some point got extract common component behaviour into an abstract class, then a refactoring must be done. Let's rewrite event processing using incapsulation instead of inherintance.

I think that if we come up with something else, we will need to refactor something.
What do you propose it will look like with incapsulation? Will it be something like: `IndexManager#eventProduser` ?

We would also need to decide where the creation of index storage (`MvTableStorage#getOrCreateIndex`) will take place, since there is a [IGNITE-19112](https://issues.apache.org/jira/browse/IGNITE-19112) to fixing this.
The manager is in the right place for me at the moment - they listen the index creation event from catalog, created index storages.

Reply:
> A new component IndexBuildController will appear, which will run the index building procedure, it will listen to events from IndexManager and PlacementDriver and react correctly to the its events.

Should IndexBuildController listen for Catalog event directly, when an index is created.

Reply:
Look what happens, in the current ticket you will delete these events, then I will return them anyway, and I will need to deal with conflicts and return tests (it is clear that new ones will need to be added). Double work (even triple work) will be done, which would be better spent on other problems and/or bugs or tasks.

Reply:
Generally speaking, it is a dead code, and it looks inappropriate at times. So let's see on our plans, and if there's no JIRA or its intentions are not clear, we may as well proceed with deleting these classes. Returning them back won't be that hard

Reply:
We discussed it personally (with @AMashenkov and @ibessonov) and decided to get rid of the events.